pkg/webhooks/workload_webhook.go

```diff
@@ -182,6 +185,103 @@ func validateContainer(c *corev1.Container, path *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateAdmissionChecks(obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i := range obj.Status.AdmissionChecks {
+		admissionChecksPath := basePath.Index(i)
+		ac := &obj.Status.AdmissionChecks[i]
+		if len(ac.PodSetUpdates) > 0 && (len(ac.PodSetUpdates) != len(obj.Spec.PodSets)) {
+			allErrs = append(allErrs, field.Invalid(admissionChecksPath.Child("podSetUpdates"), field.OmitValueType{}, "must have the same number of podSetUpdates as the podSets"))
+		}
+		allErrs = append(allErrs, validatePodSetUpdates(ac, obj, admissionChecksPath.Child("podSetUpdates"))...)
+	}
+	return allErrs
+}
+
+func validatePodSetUpdates(acs *kueue.AdmissionCheckState, obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	knowPodSets := sets.New(slices.Map(obj.Spec.PodSets, func(ps *kueue.PodSet) string {
+		return ps.Name
+	})...)
+
+	for i := range acs.PodSetUpdates {
+		psu := &acs.PodSetUpdates[i]
+		psuPath := basePath.Index(i)
+		if found := knowPodSets.Has(psu.Name); !found {
+			allErrs = append(allErrs, field.NotSupported(psuPath.Child("name"), psu.Name, knowPodSets.UnsortedList()))
+		}
+		allErrs = append(allErrs, validateTolerations(psu.Tolerations, psuPath.Child("tolerations"))...)
+		allErrs = append(allErrs, apivalidation.ValidateAnnotations(psu.Annotations, psuPath.Child("annotations"))...)
+		allErrs = append(allErrs, metav1validation.ValidateLabels(psu.Labels, psuPath.Child("labels"))...)
```

Comment:
Can we validate nodeSelector, as well?

https://github.com/kubernetes/kubernetes/blob/57d3cc66050545d7fefb0841e750a0a41df14144/pkg/apis/core/validation/validation.go#L4218-L4219

Reply:
The nodeSelector is much simpler than that. We can just use `ValidateLabels` again.

Reply:
It makes sense. We don't need to validate `operator`.